[KSSWSept20] iP

KSSWSept20 opened the pull request:
# Tes
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. Add your tasks.
2. Let it manage your tasks for you :sunglasses:

And it is **FREE**!

Features:
- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Tes chatbot = new Tes();
        chatbot.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

KSSWSept20 committed:
Changed name from Duke to Tes

KSSWSept20 committed:
Echo

KSSWSept20 committed:
Improve on Level-1. use a new class "Ui" to perform interface interaction

KSSWSept20 committed:
first version of level 2

KSSWSept20 committed:
Add, List

KSSWSept20 committed:
Added Task class. Fix the Level-2 bug. Change storage to be storing a list of Task, instead of a list of String.

KSSWSept20 committed:
Mark as done

KSSWSept20 committed:
ToDos, Events, Deadlines

KSSWSept20 committed:
Automated Text UI Testing

KSSWSept20 committed:
Handle Errors

KSSWSept20 committed:
Storage class: use ArrayList<Task> to store the tasks now instead of Task[100]

KSSWSept20 committed:
Delete

KSSWSept20 committed:
Add parser.java

Instead of parsing the commands in Tes.java, add another parser class to
deal with parsing commands.

KSSWSept20 committed:
Enable saving of tasks in the hard disk

Implementing the feature of saving the task list into a text file in the
hard disk and also loading the task list from the respective file when
running the chatbot next time.

The format of the line shown when a task is stored has been changed to
ease the implementation of this feature.

KSSWSept20 committed:
Merge branch 'branch-Level-7'

KSSWSept20 committed:
Change the way the chatbot dealing with date

The chatbot now accept date in the format yyyy-mm-dd and
print in a different format MMM dd yyyy.

The task list now stores tasks with deadline dates as
java.time.LocalDate instead of string.

KSSWSept20 committed:
Fix the bug for Event and Deadline tasks

Change the loadFromFile method in Storage.java to fix the bug where
Event and Deadline will not be loaded correctly when rerun the chatbot.

KSSWSept20 committed:
Include mechanism to deal with time precise to minute

Previously, the chatbot can only record the date for tasks of type Event
or Deadline. Now, the chatbot will record time too precise to minute.

KSSWSept20 committed:
Add and fix the comments

Add comments for every class, method and class member according to Java
SE-EDU Java Coding Standard.

Fix some comments written before which violated Java SE-EDU Java Coding
Standard.

KSSWSept20 committed:
Merge branch 'branch-Level-8'

KSSWSept20 committed:
Add TaskList.java

Seperate the function of adding, editing and deleting tasks from
Storage class. Now, the storage class only handle saving and loading
the task list.

A new class, TaskList, is created to deal with adding, editing and
deleting tasks.

KSSWSept20 committed:
Merge branch 'branch-A-MoreOOP'

KSSWSept20 committed:
Divide class into packages

Put Tes class which is the main class to run in tes package. Add two
sub packages, command and taskmanager package under tes package.

Put Parser class, Ui class and TesException class in tes.command
package. Put Task class, TaskList class, Storage class, ToDo class,
Deadline class and Event class in tes.taskmanager package.

KSSWSept20 committed:
Merge branch 'branch-A-Packages'

KSSWSept20 committed:
Merge branch 'add-gradle-support'

KSSWSept20 committed:
Add UiTest.java and TaskListTest.java

Use gradle to test the greet() method in Ui.java and the storeDeadline()
method in the TaskList.java

KSSWSept20 committed:
Add UiTest.java and TaskListTest.java

Test the greet() method in Ui.java and the storeDeadline() method in
TaskList.java.

KSSWSept20 committed:
Merge branch 'branch-A-JUnit'

KSSWSept20 committed:
Package the app as a JAR File

Change the plugin in the build.gradle file accordingly. Then, use
Shadow Gradle plugin to package the app into a JAR file.

KSSWSept20 committed:
Merge branch 'branch-A-Jar'

KSSWSept20 committed:
Add comments on testing classes

Add comments on the UiTest class and the testGreet function inside.

Add comments on the TaskListTest class and the testStoreDeadline
function inside.

KSSWSept20 committed:
Modify the code to follow coding standard

Change some parts of code in Ui class to follow the given coding
standard.

KSSWSept20 committed:
Add find function to the chatbot

Users can now find tasks easily using a keyword. A list of tasks
which contain the keyword will be shown.

KSSWSept20 committed:
Merge branch 'branch-A-JavaDoc'

KSSWSept20 committed:
Merge branch 'branch-A-CodingStandard'

KSSWSept20 committed:
Merge branch 'branch-Find'

Joseph31416 commented on src/main/java/tes/command/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        this.ui = ui;
+    }
+
+    /**
```
I like the detailed JavaDocs

Joseph31416 commented on src/main/java/tes/taskmanager/Storage.java:
```diff
@@ -0,0 +1,107 @@
+                        }
+                        storeList.add(store);
+                    } else {
+                        String[] split = line.split(" \\| ");
```
I think a better naming convention would be to replace `split` with a plural form of a noun, perhaps something like `segments`.

Joseph31416 reviewed (commented):
Overall, good code quality with rare slip ups (see specific code comments for more details).
I really like the detailed JavaDocs provided for some classes. Well done!

irwynliong commented on src/main/java/tes/taskmanager/Task.java:
```diff
@@ -0,0 +1,51 @@
+ */
+public class Task {
+    private String description; // Description of the task.
+    private boolean isDone; // Status of the task.
```
Good boolean variable naming. 😄

irwynliong reviewed (commented):
Overall very clean code.
Simple and correct!

KSSWSept20 committed:
Change branch-Find to branch-Level-9

KSSWSept20 committed:
Add checkstyle.xml and suppressions.xml

Create a new directory config/checkstyle and create these two files
inside this directory.

Use gradle to do the checkstyle and fix any violation found.

KSSWSept20 committed:
Merge branch 'branch-A-CheckStyle'

KSSWSept20 committed:
Update build.gradle

This step include lines needed to use JavaFx in the project.

KSSWSept20 committed:
Add MainWindow.fxml and DialogBox.fxml and respective java classes

These are just rough incomplete drafts. Committing this before trying
to modify the original Tes class and Ui class.

KSSWSept20 committed:
Add a dummy method to make the draft work

Add a dummy method "getResponse" to try the draft implementation.

KSSWSept20 committed:
Change the Ui and Parser class to fit GUI implementation

In the Ui.java, instead of using system.println.out to print lines,
a string is returned.

There are still some bugs on design need to be fixed.

KSSWSept20 committed:
Adjust the layout in fxml files

Remove the prefHeight of dialogContainer in the ScrollPane in
MainWindow.fxml to avoid truncating the dialog when there are many
lines.

KSSWSept20 committed:
Finalize the implementation of GUI chatbot

Add the greeting method during initializing of the chatbot and the
closing method for exiting the chatbot.

KSSWSept20 committed:
Add comment to new methods and classes

Modify comments for some old methods too as the implementation
changes with GUI added.

Delete unwanted old codes and comments.

KSSWSept20 committed:
Delete unwanted codes

Delete scanner related codes in the Ui class.

KSSWSept20 committed:
Modify layout of GUI

Use padding on Label instead of hard typing whitespaces on the code.

KSSWSept20 committed:
Merge branch 'branch-Level-10'

KSSWSept20 committed:
Generate tes.jar

KSSWSept20 committed:
Add assertion to Storage.java

In the saveToFile() method, since the tasks passed in as parameter
to be saved should not be null, an assertion is added to verify this
assumption.

In the loadFromFile() method, a record of "todo" should be split into
3 parts. A record of "deadline" or "event" should be split into 4 parts.
Thus, assertions are added to verify this assumption.

KSSWSept20 committed:
Rename and delete some items to improve code readability

Fix the comment of the constructor in Deadline.java.

Delete redundant comments on fields in Task.java.

Delete unwanted codes and comments in Storage.java. Rename variables in
Storage.java to have more meaningful names.

KSSWSept20 committed:
Change names of variables in TaskList.java to improve code readability

KSSWSept20 committed:
Rename and delete some items to improve code readability

Delete redundant comments in MainWindow.java.

Delete redundant comments in Ui.java. Rename a variable in Ui.java to
become more appropriate with its intended role.

Rename variables in parse() method in Parser.java to have a more
meaningful name for each variable.

KSSWSept20 committed:
Merge pull request #2 from KSSWSept20/branch-A-Assertions

Add assertion to Storage.java

KSSWSept20 committed:
Merge branch 'master' into branch-A-CodeQuality

KSSWSept20 committed:
Merge pull request #3 from KSSWSept20/branch-A-CodeQuality

Branch a code quality

KSSWSept20 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data/Tes.txt
#	src/main/java/tes/taskmanager/Storage.java

KSSWSept20 committed:
Test the chatbot by running a few command

KSSWSept20 committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/KSSWSept20/ip
into branch-A-CodeQuality

# Conflicts:
#	data/Tes.txt

KSSWSept20 committed:
Add user guide function to the chatbot

As a user guide is useful for new users, users can refer to the guide
for formats of commands. A new help Window will be shown when the user
guide is shown.

Only a skeleton of the Help Window is built at this point.

KSSWSept20 committed:
Create the real user guide for chatbot

Use table to show a summary of commands and the respective formats.
Examples are also given to show a clear guide to the user.

KSSWSept20 committed:
Resize the table of User Guide to give a better visual

KSSWSept20 committed:
Merge branch 'branch-C-Help'

KSSWSept20 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	data/Tes.txt

KSSWSept20 committed:
Set a title for the chatbot

KSSWSept20 committed:
Improve the visual of dialogboxes

Add background color to the dialogue box to appear similar as a chat
bubble. User's dialogue box is in light blue color while the chatbot's
dialogue box is in light grey color.

Attempt to crop the profile image to a circle and try to add a circular
border to it but still getting a weird result now.

KSSWSept20 committed:
Add an image for background of chatbot

KSSWSept20 committed:
Change background color instead of using a background image

As changing the background using a background image is more complicated
and may involve the usage of CSS, I decided to just change the
background color of the chatbot.

KSSWSept20 committed:
Delete redundant code in DialogBox.fxml

KSSWSept20 committed:
Crop the image to become a square image

This can ease the implementation of cropping the image into a circle.

KSSWSept20 committed:
Change font and size of each dialogue

KSSWSept20 committed:
Make the scrollpane and text field resizable

Intended to make the dialogbox to be resizable at first, but the
implementation is more complicated than expected so I have to give up
due to time constraint.

Will try to implement again in the future.

KSSWSept20 committed:
Merge branch 'branch-A-BetterGui'

KSSWSept20 committed:
Cut the long method in Parser class to meet SLAP

KSSWSept20 committed:
Change the response text to add personality to the chatbot

KSSWSept20 committed:
Merge branch 'branch-A-Personality'

KSSWSept20 committed:
Throw exception when an out-of-bound or non-positive integer is passed
in to the parameter of mark, unmark or delete method

KSSWSept20 committed:
Merge branch 'branch-A-MoreErrorHandling'

KSSWSept20 committed:
Give credit to gui-related file

KSSWSept20 committed:
Change the profile picture of the chatbot

KSSWSept20 committed:
Fix a test method in UiTest.java

Also added another method for testing close method in Ui.java.

KSSWSept20 committed:
Fix a bug in mark and unmark method

KSSWSept20 committed:
Add Ui.png to docs

KSSWSept20 committed:
Modify the loadFromFile method in Storage.java to meet SLAP

Create helper functions to shorten the loadFromFile method.

KSSWSept20 committed:
Add comments to some helper methods

KSSWSept20 committed:
Add a screenshot of the Help Window

KSSWSept20 committed:
Add a screenshot of demonstration of find method

KSSWSept20 committed:
Update README.md

KSSWSept20 committed:
Add more error handling and method testing

KSSWSept20 committed:
Include dependencies for Mac and Linux

KSSWSept20 committed:
Fix the codes according to Checkstyle
